### PR TITLE
Agent refactor

### PR DESCRIPTION
**Pull request by spomichter**
Agent System Refactor: Unified Gateway & Enhanced Conversation Management

  Summary

  Major refactor of the DimOS agent system to use TensorZero as a unified LLM gateway, eliminating provider-specific code
  while adding robust conversation history management and standardized message types.

  Key Changes

  🏗️ Architecture Improvements

  - Unified Gateway: All LLM providers (OpenAI, Claude, Cerebras, Qwen) now route through UnifiedGatewayClient with
  TensorZero embedded gateway
  - BaseAgent Hierarchy: Clean inheritance with BaseAgent (standalone) and BaseAgentModule (DimOS module wrapper)
  - Removed Provider-Specific Code: Deleted claude_agent.py, cerebras_agent.py, old agent.py - all functionality unified

  💬 Conversation History Management

  - NEW: ConversationHistory Type: Thread-safe conversation manager with proper locking and format conversion
  - Fixed Critical Bug: Agent now properly remembers previous messages across queries
  - OpenAI Format Compliance: Messages maintain exact format required by OpenAI/TensorZero APIs
  - Multimodal Support: Handles text, images, and tool calls in conversation history

  📦 Standardized Message Types

  - AgentMessage: Unified input type for text + images with pickling support for module communication
  - AgentResponse: Standardized output with content, tool_calls, and metadata
  - AgentImage: Base64 JPEG encoded images optimized for LLM processing
  - Image.agent_encode(): New method for automatic RGB conversion and encoding

  🧪 Comprehensive Test Suite

  - 7 Production Test Files: Full coverage for text, vision, tools, streaming, and conversation history
  - Mock Tests for CI/CD: All tests have mock versions that don't require API keys
  - Parametrized Provider Tests: Automatic testing across all supported providers
  - New Test: test_conversation_history.py with edge cases and thread safety tests

  Breaking Changes

  - Modules now use AgentMessage for input (no more separate query/image/data streams)
  - Modules output AgentResponse objects (not raw strings)
  - agent.history replaced with agent.conversation (ConversationHistory type)

### CODE REVIEW

File: dimos/agents/modules/base_agent.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# Copyright 2025 Dimensional Inc.
```
Comment (spomichter) [inline dimos/agents/modules/base_agent.py:1]: Rename this to base agent module

File: dimos/agents/modules/agent_pool.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
+# Copyright 2025 Dimensional Inc.
```
Comment (spomichter) [inline dimos/agents/modules/agent_pool.py:1]: In development

File: dimos/agents/modules/agent.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
+# Copyright 2025 Dimensional Inc.
```
Comment (spomichter) [inline dimos/agents/modules/agent.py]: Not used, deprecated. Delete
